pkg/pulumiyaml/run.go

```diff
@@ -1230,6 +1255,27 @@ func (e *programEvaluator) registerResource(kvp resourceNode) (lateboundResource
 	if v.Options.PluginDownloadURL != nil {
 		opts = append(opts, pulumi.PluginDownloadURL(v.Options.PluginDownloadURL.Value))
 	}
+
+	if v.Options.Version != nil {
+		versionValue, ok := e.evaluateExpr(v.Options.Version)
+		if ok {
+			if !hasOutputs(versionValue) {
+				if version, ok := versionValue.(string); ok {
+					opts = append(opts, pulumi.Version(version))
+				} else {
+					e.error(v.Options.Version, "version must be a string value")
+					overallOk = false
+				}
+			} else {
+				e.error(v.Options.Version, "version must be not be an output")
```

Comment:
I don't remember if the type checker can catch this kind of thing, but if it can that would be great.